### PR TITLE
Feat/stacks single node follower

### PR DESCRIPTION
**Pull request by AshtonStephens**
### What does this PR do?

> *Note: This is part of a series of iterative but bulky PRs into `stacks-network/aws-blockchain-node-runners feat/stacks-node` before creating a PR into `aws-samples/aws-blockchain-node-runners` from that feature branch. After we've reviewed the code here and have the full documented node with all the node types we plan to support we can open a PR to the `aws-samples/aws-blockchain-node-runners` upstream fork.*

Changes:

1. Copy the Solana node code to Stacks
2. Change names to use Stacks
3. Alter template to make a Stacks node
4. *etc.* TODO: fill in later.

### Motivation

GH Issues:
- Stacks Core: https://github.com/stacks-network/stacks-core/issues/4325
- AWS Node Runners: https://github.com/aws-samples/aws-blockchain-node-runners/issues/42

### More

> *Note: I am skipping some of the documentation in this initial PR because I plan to add other node types after this, and it makes more sense to write the documentation for all of them at the same time.*

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Mandatory for new node blueprints. Yes, I have added usage example to the `README.md` file in my blueprint folder
- [x] Mandatory for new node blueprints. Yes, I have implemented automated tests for all stacks in my blueprint and they pass
- [ ] Mandatory for new node blueprints. Yes, I have added a reference to my `README.md` file to `website/docs` section for this feature
- [x] Yes, I have set up and ran all [pre-merge quality control tools](./docs/pre-merge-tools.md) on my local machine and they don't show warnings.

### For Moderators

- [ ] The tests for all current blueprints successfully complete before merge?
- [ ] Mandatory for new node blueprints. All [pre-merge quality control tools](./docs/pre-merge-tools.md) and `cdk-nag` tools don't show warnings?
- [ ] Mandatory for new node blueprints. The deployment test works on blank AWS account according to instructions in the `README.md` before merge?
- [ ] Mandatory for new node blueprints. The website builds without errors?

#### Dashboard for a node runner that spun up with no outside input.
Screenshot 2024-02-16 at 6 58 31 PM

### CODE REVIEW

**Comment by JakeBlockchain** [comment]
@AshtonStephens What are the stats for the machine running the stacks-single-nodefollower above?
I'd like to compare to my current local machine.

**Comment by AshtonStephens** [comment]
It's the defaults, but run with this:
```bash
#############################################################
# Example configuration for Stacks nodes runner app on AWS  #
#############################################################

## Set the AWS account is and region for your environment ##
AWS_ACCOUNT_ID="xxxxxxxxxxxx"
AWS_REGION="us-west-2"

## Common configuration parameters ##
STACKS_NETWORK="mainnet"                               # All options:  "mainnet", "testnet"
STACKS_NODE_CONFIGURATION="follower"                   # All options:   "follower", "signer", "miner"
STACKS_VERSION="latest"                             # Current required version of Stacks

STACKS_INSTANCE_TYPE="m5.large"
STACKS_CPU_TYPE="x86_64"                      # All options: "x86_64", "ARM_64". IMPORTANT: Make sure the CPU type matches the instance type used

# Secrets configuration
STACKS_SIGNER_SECRET_ARN="none"                            # Optional. The ARN of the secret in AWS Secrets Manager with the signer private key in it.
STACKS_MINER_SECRET_ARN="none"                             # Optional. The ARN of the secret in AWS Secrets Manager with the miner private key in it.

# Bitcoin configuration
BITCOIN_PEER_HOST=bitcoind.stacks.co
BITCOIN_RPC_USERNAME=blockstack
BITCOIN_RPC_PASSWORD=blockstacksystem

# Stacks configuration
STACKS_BOOTSTRAP_NODE="02196f005965cebe6ddc3901b7b1cc1aa7a88f305bb8c5893456b8f9a605923893@seed.mainnet.hiro.so:20444,02539449ad94e6e6392d8c1deb2b4e61f80ae2a18964349bc14336d8b903c46a8c@cet.stacksnodes.org:20444,02ececc8ce79b8adf813f13a0255f8ae58d4357309ba0cedd523d9f1a306fcfb79@sgt.stacksnodes.org:20444,0303144ba518fe7a0fb56a8a7d488f950307a4330f146e1e1458fc63fb33defe96@est.stacksnodes.org:20444"
```

**Comment by AshtonStephens** [comment]
m5.large: https://instances.vantage.sh/aws/ec2/m5.large

**Comment by AshtonStephens** [comment]
Discussed this with the rest of the team, for now I'm going to be merging these myself and will seek more input on the final merge in.

File: lib/stacks/lib/assets/user-data/node.sh [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+sleep 60
+
+sudo mkdir -p $DATA_VOLUME_PATH
+
+if [[ "$DATA_VOLUME_TYPE" == "instance-store" ]]; then
+  echo "Data volume type is instance store"
+
+  sudo chmod +x $CLOUD_ASSETS_PATH/setup-instance-store-volumes.sh
+
+  (crontab -l; echo "@reboot $CLOUD_ASSETS_PATH/setup-instance-store-volumes.sh > /tmp/setup-instance-store-volumes.log 2>&1") | crontab -
```
Comment (wileyj) [inline lib/stacks/lib/assets/user-data/node.sh:126]: had you considered a systemd timer unit here vs crontab? nothing wrong with cron, just curious

**Comment by wileyj** [comment]
Another comment re: shell in the files here: i usually run any shell through a tool like this: https://github.com/koalaman/shellcheck

there are a few lines you're calling that may have unintended consequences if a variable isn't set (or overwritten), ex: `mkdir -p $VAR/bin` 

for lines like that, shellcheck will complain that it's not quoted:
```
In test line 2:
mkdir -p $VAR/bin
         ^--^ SC2086 (info): Double quote to prevent globbing and word splitting.

Did you mean: 
mkdir -p "$VAR"/bin

For more information:
  https://www.shellcheck.net/wiki/SC2086 -- Double quote to prevent globbing ...
```


overall, looks great - just some minor nits we can look into for the next PR

File: lib/stacks/lib/assets/user-data/node.sh (hunk 1 above)
Comment (frbrkoala) [reply lib/stacks/lib/assets/user-data/node.sh:126]: No, TBH I just used whatever I am more familiar with.

**Pull request merged**